src/DssSpell.t.base.sol

```diff
@@ -1410,57 +1414,51 @@ contract DssSpellTestBase is Config, DssTest {
         );
     }
 
+    /**
+     * @dev Checks if the deployer of a contract has not kept `wards` access to the contract.
+     * Notice that it depends on `deployers` being kept up-to-date.
+     */
     function _checkWards(address _addr, string memory contractName) internal {
         for (uint256 i = 0; i < deployers.count(); i ++) {
             address deployer = deployers.addr(i);
-            (bool ok, bytes memory data) = _addr.call(
-                abi.encodeWithSignature("wards(address)", deployer)
-            );
+            if(_skipWards(_addr, deployer)) {
+                continue;
+            }
+
+            (bool ok, bytes memory data) = _addr.call(abi.encodeWithSignature("wards(address)", deployer));
             if (!ok || data.length != 32) return;
+
             uint256 ward = abi.decode(data, (uint256));
             if (ward > 0) {
-                if (_skipWards(_addr, deployer)) continue;
-                emit log("Error: Bad Auth");
                 emit log_named_address("   Deployer Address", deployer);
                 emit log_named_string("  Affected Contract", contractName);
-                fail();
+                fail("Error: Bad Auth");
             }
         }
     }
 
-    function _checkSource(address _addr, string memory contractName) internal {
-        (bool ok, bytes memory data) =
-            _addr.call(abi.encodeWithSignature("src()"));
+    /**
+     * @dev Same as `_checkWards`, but for OSMs' underlying Median contracts.
+     */
+    function _checkOsmSrcWards(address _addr, string memory contractName) internal {
+        (bool ok, bytes memory data) = _addr.call(abi.encodeWithSignature("src()"));
         if (!ok || data.length != 32) return;
+
         address source = abi.decode(data, (address));
-        string memory sourceName = string(
-            abi.encodePacked("source of ", contractName)
-        );
+        string memory sourceName = string(abi.encodePacked("src of ", contractName));
```

Comment:
On the mainnet [you've used](https://github.com/makerdao/spells-mainnet/blob/9c25724f55513c117655ee31ebee240bd7d97287/src/DssSpell.t.base.sol#L1518) newly updated `_concat` function instead